### ISSUE TITLE
Add insights into the processing

### ISSUE THREAD

**Issue by rosell-dk** [rosell-dk/webp-convert#49]
Right now, `WebPConvert::convert` is a black box. It is not possible to get insight into which converters succeeded, and which failed, and why they failed. In order to be able to get that kind of information, you would have to practically duplicate the functionality of `WebPConvert::convert`, adding logging.

It would be practical if WebPConvert::convert were able to do the logging. I cannot imagine a serious implementation using WebPConvert, which does not need to provide the user insight into the box

**Comment by rosell-dk** [comment]
I suggest that WebPConvert optionally takes a logger object, and calls a "log()" method on that object

**Comment by rosell-dk** [comment]
WebPConvert could provide a "EchoLogger", which simply echoes the log messages, and a "VoidLogger", which does nothing.

Example usage:
```
$result = WebPConvert::convert($source, $destination, array(
    'quality' => 80,
    'metadata' => 'none',       // Valid values: all, none, exif, icc, xmp
    'method' => 6,
    'converters' => array(
        array(
            'converter' => 'ewww',
            'options' => array(
                'key' => '',
            ),
        ),
        'cwebp'
    )
), new \WebPConvert\Loggers\EchoLogger());
```

**Comment by rosell-dk** [comment]
Fixed in f85390a

**Comment by rosell-dk** [comment]
Its great, but I want more!
I would like insights from the individual converters as well
